common/src/main/java/com/wynntils/features/user/overlays/map/MiniMapOverlayFeature.java

```diff
@@ -100,33 +109,130 @@ public void render(PoseStack poseStack, float partialTicks, Window window) {
-
-            // disable rotation if necessary
-            if (followPlayerRotation) {
-                poseStack.popPose();
+            if (WebManager.isMapLoaded()) {
+                renderMapQuad(map, poseStack, centerX, centerZ, textureX, textureZ, width, height);
             }
 
             // TODO minimap icons
 
             // TODO compass icon
 
             // cursor
+            renderCursor(poseStack, centerX, centerZ);
+
+            // disable mask & render border
+            switch (maskType) {
+                case Rectangular -> {
+                    RenderSystem.disableScissor();
+                    renderRectangularMapBorder(poseStack, renderX, renderY, width, height);
+                }
+                    // case Circle -> {
+                    // TODO
+                    // renderCircularMapBorder();
+                    // }
+            }
+
+            // Directional Text
+            renderCardinalDirections(poseStack, width, height, centerX, centerZ);
+
+            // Coordinates
+            if (showCoords) {
+                FontRenderer.getInstance()
+                        .renderText(
+                                poseStack,
+                                String.format(
+                                        "%s, %s, %s",
+                                        (int) McUtils.player().getX(),
+                                        (int) McUtils.player().getY(),
+                                        (int) McUtils.player().getZ()),
+                                centerX,
+                                renderY + height + 6,
+                                CommonColors.WHITE,
+                                HorizontalAlignment.Center,
+                                VerticalAlignment.Middle,
+                                FontRenderer.TextShadow.NONE);
+            }
+        }
+
+        private void renderCardinalDirections(
+                PoseStack poseStack, float width, float height, float centerX, float centerZ) {
+            if (showCompass != CompassRenderType.None) {
+                float northDX;
+                float northDY;
+
+                if (followPlayerRotation) {
+                    float yawRadians = (float) Math.toRadians(McUtils.player().getYRot());
+                    northDX = (float) StrictMath.sin(yawRadians);
+                    northDY = (float) StrictMath.cos(yawRadians);
+                    if (maskType == MapMaskType.Rectangular) {
+                        // Scale as necessary
+                        double toSquareScale = Math.min(width / Math.abs(northDX), height / Math.abs(northDY)) / 2;
+                        northDX *= toSquareScale;
+                        northDY *= toSquareScale;
+                    }
+
+                } else {
+                    northDX = width / 2;
+                    northDY = height / 2;
+                }
+
+                FontRenderer.getInstance()
```

Comment:
Perhaps a local method to de-duplicate code (and pass renderX, renderY and text)?

Reply:
Should be partially lowered through using a render setting now